aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -158,7 +165,31 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       }
     } finally {
       cleanupMetadataAndUnlock(commitStatus, newMetadataLocation);
+      if (glueTempTableCreated && commitStatus != CommitStatus.SUCCESS) {
+        glue.deleteTable(DeleteTableRequest.builder().databaseName(databaseName).name(tableName).build());
+      }
+    }
+  }
+
+  private boolean createGlueTempTable(TableMetadata base, String metadataLocation) {
+    try {
+      if (awsProperties.glueLakeFormationEnabled() && base == null) {
+        // LakeFormation credential require TableArn as input，so creating a dummy table
+        // beforehand for create table scenario
+        glue.createTable(CreateTableRequest.builder()
+            .databaseName(databaseName)
+            .tableInput(TableInput.builder()
+                .parameters(ImmutableMap.of(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE))
+                .name(tableName)
+                .storageDescriptor(StorageDescriptor.builder().location(metadataLocation).build())
+                .build())
+            .build());
+        return true;
+      }
+    } catch (GlueException e) {
+      return false;
```

Comment:
I think it should not mute the exception, the operation should fail if temp table creation fails